packages/gatsby/src/commands/build.ts

```diff
@@ -233,10 +233,7 @@ module.exports = async function build(program: IBuildArgs): Promise<void> {
   workerPool.end()
   buildActivity.end()
 
-  if (
-    process.env.GATSBY_EXPERIMENTAL_PAGE_BUILD_ON_DATA_CHANGES &&
-    process.argv.includes(`--log-pages`)
-  ) {
+  if (process.argv.includes(`--log-pages`)) {
```

Comment:
shouldn't we move this to the cli? 😬 
Can you create a follow up ticket?

Reply:
Ah, we probably already get things on `program` for it because we do let `yargs` know about them (just hide them from `--help`) - https://github.com/gatsbyjs/gatsby/blob/1048eda5d3c0ae16c55c3296280359bff559488f/packages/gatsby-cli/src/create-cli.ts#L246-L261

I'll create follow up ticket on that